src/backends/fvm_gpu.hpp

```diff
@@ -51,6 +53,14 @@ __global__ void matrix_solve(matrix_solve_param_pack<T, I> params);
 template <typename T, typename I>
 __global__ void assemble_matrix(matrix_update_param_pack<T, I> params, T dt);
 
+template <typename T, typename I, typename Stack>
+__global__
+void test_thresholds(
+    float t, float t_prev, int size,
```

Comment:
Sorry its not clear to me what t is here: time, threshold, template.
It appears an important function, maybee add function description